### PR TITLE
[CLEANUP] Remove the things we EOL'd in 2.6.0

### PR DESCRIPTION
**Pull request by chadhietala**
This removes items that were EOL'd in 2.6.0. Those things are:
- Legacy View Support
- Legacy Controller Support

As part of this I found many tests that have been ported to the new test infra but were never deleted. Those are also included as they were using `View`s for testing.

### CODE REVIEW

File: packages/ember-htmlbars/lib/hooks/bind-self.js [hunk 1]
```diff
@@ -2,14 +2,10 @@
 @module ember
 @submodule ember-htmlbars
 */
-import { ENV } from 'ember-environment';
 import ProxyStream from '../streams/proxy-stream';
 
 export default function bindSelf(env, scope, self) {
   if (self && self.isView) {
```
Comment (chadhietala) [inline packages/ember-htmlbars/lib/hooks/bind-self.js]: Perhaps this can go further?

Comment (rwjblue) [inline packages/ember-htmlbars/lib/hooks/bind-self.js]: Yes, lets try to kill this whole `self && self.isView` block.

File: packages/ember-metal/lib/index.js [hunk 2]
```diff
@@ -275,30 +275,6 @@ Ember.isGlobalPath = isGlobalPath;
...
-@for Ember
-@private
-*/
-Ember.Backburner = function() {
-  deprecate(
-    'Usage of Ember.Backburner is deprecated.',
-    false,
-    {
-      id: 'ember-metal.ember-backburner',
-      until: '2.8.0',
```
Comment (rwjblue) [inline packages/ember-metal/lib/index.js]: This one needs to stay.  `until` is inclusive (meaning we have to keep it around through 2.8.0).  Due to the LTS policy this needs to stick around until we release another LTS version (which will be 2.8).

Comment (chadhietala) [inline packages/ember-metal/lib/index.js]: Were you going to cherry pick this to the beta branch? The reason why I removed it is master should be the next LTS.

Comment (rwjblue) [inline packages/ember-metal/lib/index.js]: The deprecation needs to be included in 2.8.0, which is what master is tracking.

File: packages/ember-htmlbars/tests/attr_nodes/boolean_test.js [hunk 3]
```diff
@@ -1,92 +0,0 @@
...
-import { compile } from '../utils/helpers';
-import { equalInnerHTML } from 'htmlbars-test-helpers';
-
-var view;
-
-function appendView(view) {
-  run(function() { view.appendTo('#qunit-fixture'); });
-}
-
-QUnit.module('ember-htmlbars: boolean attribute', {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/attr_nodes/boolean_test.js]: This is now tested by [these tests](https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/tests/integration/input-test.js)

File: packages/ember-htmlbars/tests/attr_nodes/data_test.js [hunk 4]
```diff
@@ -1,258 +0,0 @@
...
-import EmberObject from 'ember-runtime/system/object';
-import { compile } from '../utils/helpers';
-import { InteractiveRenderer } from 'ember-htmlbars/renderer';
-import { equalInnerHTML } from 'htmlbars-test-helpers';
-import { domHelper as dom } from 'ember-htmlbars/env';
-import { runAppend, runDestroy } from 'ember-runtime/tests/utils';
-
-var view, originalSetAttribute, setAttributeCalls, renderer;
-
-QUnit.module('ember-htmlbars: data attribute', {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/attr_nodes/data_test.js]: These are now tested by [these tests](https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/tests/integration/content-test.js#L385)

File: packages/ember-htmlbars/tests/attr_nodes/href_test.js [hunk 5]
```diff
@@ -1,29 +0,0 @@
...
-
-QUnit.module('ember-htmlbars: href attribute', {
-  teardown() {
-    if (view) {
-      run(view, view.destroy);
-    }
-  }
-});
-
-QUnit.test('href is set', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/attr_nodes/href_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13626/files#diff-e493f9c2deceaafd80b881cfabc92186R620)

File: packages/ember-htmlbars/tests/attr_nodes/property_test.js [hunk 6]
```diff
@@ -1,67 +0,0 @@
...
-}
-
-function canSetFalsyMaxLength() {
-  var input = document.createElement('input');
-  input.maxLength = 0;
-
-  return input.maxLength === 0;
-}
-
-QUnit.module('ember-htmlbars: property', {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/attr_nodes/property_test.js]: These are now tested [here](https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/tests/integration/input-test.js#L52)

File: packages/ember-htmlbars/tests/attr_nodes/sanitized_test.js [hunk 7]
```diff
@@ -1,125 +0,0 @@
...
-
-import EmberView from 'ember-views/views/view';
-import { compile } from '../utils/helpers';
-import { SafeString } from 'ember-htmlbars/utils/string';
-import { runDestroy } from 'ember-runtime/tests/utils';
-import { environment } from 'ember-environment';
-
-var view;
-
-QUnit.module('ember-htmlbars: sanitized attribute', {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/attr_nodes/sanitized_test.js]: These are now tested [here](https://github.com/emberjs/ember.js/blob/3cfc9614bbd92ee8e80b6c4894b024f2e14bffae/packages/ember-glimmer/tests/integration/helpers/unbound-test.js#L98)

File: packages/ember-htmlbars/tests/attr_nodes/svg_test.js [hunk 8]
```diff
@@ -1,73 +0,0 @@
...
-import { compile } from '../utils/helpers';
-import { equalInnerHTML } from 'htmlbars-test-helpers';
-
-var view;
-
-function appendView(view) {
-  run(function() { view.appendTo('#qunit-fixture'); });
-}
-
-QUnit.module('ember-htmlbars: svg attribute', {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/attr_nodes/svg_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13626/files#diff-11b72e1055224bd943daa20ffecc8dc1R1)

File: packages/ember-htmlbars/tests/attr_nodes/value_test.js [hunk 9]
```diff
@@ -1,53 +0,0 @@
...
-import run from 'ember-metal/run_loop';
-import { compile } from '../utils/helpers';
-
-var view;
-
-function appendView(view) {
-  run(function() { view.appendTo('#qunit-fixture'); });
-}
-
-QUnit.module('ember-htmlbars: value attribute', {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/attr_nodes/value_test.js]: Now tested [here](https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/tests/integration/input-test.js#L6)

File: packages/ember-views/tests/views/view_test.js [hunk 10]
```diff
@@ -41,31 +41,9 @@ QUnit.test('should not add role attribute unless one is specified', function() {
   ok(view.$().attr('role') === undefined, 'does not have a role attribute');
 });
 
-QUnit.test('should allow tagName to be a computed property [DEPRECATED]', function() {
```
Comment (chadhietala) [inline packages/ember-views/tests/views/view_test.js]: Was set for deprecation in 2.0.0

File: packages/ember-views/tests/views/view_test.js [hunk 11]
```diff
@@ -41,31 +41,9 @@ QUnit.test('should not add role attribute unless one is specified', function() {
...
-    view.set('tagName', 'div');
-  });
-
-  equal(view.element.tagName, 'SPAN', 'the tagName cannot be changed after initial render');
-});
-
 import { test } from 'internal-test-helpers/tests/skip-if-glimmer';
 
-test('should re-render if the context is changed', function() {
+QUnit.skip('should re-render if the context is changed', function() {
```
Comment (chadhietala) [inline packages/ember-views/tests/views/view_test.js]: This is specifically testing context and we are removing the implementation of context in this PR.

File: packages/ember-metal/lib/index.js (hunk 2 above)
Comment (chadhietala) [inline packages/ember-metal/lib/index.js]: Ah ok. I will revert this.

File: packages/ember-views/tests/views/view_test.js (hunk 11 above)
Comment (chadhietala) [inline packages/ember-views/tests/views/view_test.js]: Maybe this test can just go away?

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 12]
```diff
@@ -390,18 +373,14 @@ QUnit.test('{{render}} helper should not leak controllers', function() {
...
-  let controller = EmberController.create({
-    [OWNER]: appInstance,
-    zero: false
-  });
+  let template = '<h1>HI</h1> {{render \'post\' zero}} {{render \'post\' nonexistent}}';
+  let component;
 
   expectDeprecation(() => {
-    view = EmberView.create({
+    component = EmberView.create({
```
Comment (rwjblue) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: Should this have changed to `Component` like the others above?

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 13]
```diff
@@ -420,37 +399,30 @@ QUnit.test('{{render}} helper should not treat invocations with falsy contexts a
...
-  });
-
-  var controller = Controller.create({
-    [OWNER]: appInstance
-  });
+  let component;
 
   expectDeprecation(() => {
-    view = EmberView.create({
+    component = EmberView.create({
```
Comment (rwjblue) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: Should this use `Component`, and swap to `layout` below?

File: packages/ember-views/tests/views/view_test.js (hunk 11 above)
Comment (rwjblue) [inline packages/ember-views/tests/views/view_test.js]: Ya, kill it.

File: packages/ember-views/tests/views/view_test.js [hunk 14]
```diff
@@ -1,81 +0,0 @@
-import { computed } from 'ember-metal/computed';
-import run from 'ember-metal/run_loop';
-import jQuery from 'ember-views/system/jquery';
-import EmberView from 'ember-views/views/view';
-import { compile } from 'ember-template-compiler';
-
-var view;
-
-QUnit.module('Ember.View', {
```
Comment (chadhietala) [inline packages/ember-views/tests/views/view_test.js]: Testing ye ol `View` as public API.

**Comment by rwjblue** [comment]
Likely needs a rebase to pass.

**Comment by rwjblue** [comment]
Thank you!

**Comment by stefanpenner** [comment]
`+288 −1,014` 👍

**Pull request merged**